java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/debugging/attach/AttachConfigurations.java

```diff
@@ -0,0 +1,216 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+
+import org.netbeans.modules.java.lsp.server.protocol.DebugConnector;
+import org.netbeans.modules.java.lsp.server.protocol.NbCodeLanguageClient;
+import org.netbeans.modules.java.lsp.server.protocol.QuickPickItem;
+import org.netbeans.modules.java.lsp.server.protocol.ShowQuickPickParams;
+import org.openide.util.RequestProcessor;
+
+/**
+ * Debugger attach configurations provider.
+ *
+ * @author Martin Entlicher
+ */
+public final class AttachConfigurations {
+
+    static final String NAME_ATTACH_PROCESS = "Attach to Process";          // NOI18N
+    static final String NAME_ATTACH_SOCKET = "Attach to Port";              // NOI18N
+    static final String NAME_ATTACH_SHMEM = "Attach to Shared Memory";      // NOI18N
+    static final String NAME_ATTACH_BY = "Attach by ";                      // NOI18N
+
+    static final String CONNECTOR_PROCESS = "com.sun.jdi.ProcessAttach";    // NOI18N
+    static final String CONNECTOR_SOCKET = "com.sun.jdi.SocketAttach";      // NOI18N
+    static final String CONNECTOR_SHMEM = "com.sun.jdi.SharedMemoryAttach"; // NOI18N
+
+    static final String PROCESS_ARG_PID = "processId";          // NOI18N
+    static final String SOCKET_ARG_HOST = "hostName";           // NOI18N
+    static final String SOCKET_ARG_PORT = "port";               // NOI18N
+    static final String SHMEM_ARG_NAME = "sharedMemoryName";    // NOI18N
+
+    private static final RequestProcessor RP = new RequestProcessor(AttachConfigurations.class);
+
+    private AttachConfigurations() {}
+
+    public static CompletableFuture<Object> findConnectors() {
+        return CompletableFuture.supplyAsync(() -> {
+            return listAttachingConnectors();
+        }, RP);
+    }
+
+    private static List<DebugConnector> listAttachingConnectors() {
+        VirtualMachineManager vmm = Bootstrap.virtualMachineManager ();
+        List<AttachingConnector> attachingConnectors = vmm.attachingConnectors();
+        List<DebugConnector> connectors = new ArrayList<>(5);
+        String type = "java8+";             // NOI18N
+        for (AttachingConnector ac : attachingConnectors) {
+            String connectorName = ac.name();
+            Map<String, Connector.Argument> defaultArguments = ac.defaultArguments();
+            DebugConnector connector;
+            switch (connectorName) {
+                case CONNECTOR_PROCESS:
+                    connector = new DebugConnector(NAME_ATTACH_PROCESS, type,
+                            Collections.singletonList(PROCESS_ARG_PID),
+                            Collections.singletonList("${command:java.attachDebugger.pickProcess}"));    // NOI18N
```

Comment:
This could eventually use `JAVA_FIND_DEBUG_PROCESS_TO_ATTACH` constant

Reply:
Good point, thanks.